src/pybind/mgr/dashboard/controllers/user.py

```diff
@@ -89,3 +89,24 @@ def set(self, username, password=None, name=None, email=None, roles=None):
         user.set_roles(user_roles)
         mgr.ACCESS_CTRL_DB.save()
         return User._user_to_dict(user)
+
+
+@ApiController('/user/{username}')
+class UserChangePassword(BaseController):
+    @Endpoint('POST')
+    def change_password(self, username, old_password, new_password):
```

Comment:
It is not possible to use ``POST /user/<:id>`` because this requires ``Scope.USER`` privileges which for example a ``read-only`` user does not have. In this case such a user is not able to change his password.

Reply:
Requiring the old password for validation is a common security thing. All services (e.g. GitHub, Jira, Trello) i know require the input of the old password when changing the password.

Reply:
> It is not possible to use `POST /user/<:id>` because this requires `Scope.USER` privileges which for example a `read-only` user does not have. In this case such a user is not able to change his password.

Mmm, interesting issue, as in this case `User` is the resource where auth rights emanate (kind of vicious circle).

Would it make sense then to have an aliased URL `/user/self` (not sure if CherryPy mapper will allow this), so all users (regardless their permissions) can interact with it:

```python
@ApiController('/user/self')
class SelfUser(User):
  def update(self, old_password, new_password):
    super(SelfUser, self).set(JwtManager.get_token_from_header()['username'], ...)
```

Reply:
This will collide with the scenario when there is a user that is called ``self``. Every endpoint that is ``/user/xxx`` except ``/user/<namename>`` can not be implemented, the parameter after ``/user`` is always a username.